e2e-tests/playwright/utils/UIhelper.ts

```diff
@@ -176,25 +180,54 @@ export class UIhelper {
   }
 
   async verifyColumnHeading(
-    rowTexts: string[] | RegExp[],
+    rowTexts: (string | RegExp | (string | RegExp)[])[],
     exact: boolean = true,
   ) {
     for (const rowText of rowTexts) {
-      const rowLocator = this.page
-        .locator(`tr>th`)
-        .getByText(rowText, { exact: exact })
-        .first();
-      await rowLocator.waitFor({ state: 'visible' });
-      await rowLocator.scrollIntoViewIfNeeded();
-      await expect(rowLocator).toBeVisible();
+      if (Array.isArray(rowText)) {
+        let found = false;
+
+        for (const text of rowText) {
+          const rowLocator = this.page
+            .locator('tr>th')
+            .getByText(text, { exact })
+            .first();
+
+          // rowLocator.allTextContents().then((res) => console.log(res));
+          // Check if the locator finds the text in the page
+          if ((await rowLocator.count()) > 0) {
+            await rowLocator.waitFor({ state: 'visible' });
+            await rowLocator.scrollIntoViewIfNeeded();
+            await expect(rowLocator).toBeVisible();
+            found = true;
+            break; // Exit the loop as soon as one match is found
+          }
+        }
+
+        if (!found) {
+          // If none of the expected texts are found, throw an error
+          throw new Error(
+            `None of the expected texts were found in column heading: ${rowText.join(', ')}`,
+          );
+        }
+      } else {
+        // If it's not an array, just check the single string as usual
+        const rowLocator = this.page
+          .locator('tr>th')
+          .getByText(rowText, { exact })
+          .first();
+        await rowLocator.waitFor({ state: 'visible' });
+        await rowLocator.scrollIntoViewIfNeeded();
+        await expect(rowLocator).toBeVisible();
+      }
     }
   }
 
-  async verifyHeading(heading: string) {
+  async verifyHeading(heading: string, exact: boolean = true) {
     const headingLocator = this.page
       .locator('h1, h2, h3, h4, h5, h6')
-      .filter({ hasText: heading })
-      .first();
+      .getByText(heading, { exact });
```

Comment:
with a same text there can be multiple Headings, lets keep `.first();` to make test less flaky.